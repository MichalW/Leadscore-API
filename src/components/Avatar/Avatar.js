import PropTypes from 'prop-types';
import React from 'react';
import get from 'lodash/get';
import upperCase from 'lodash/upperCase';
import {Image} from 'semantic-ui-react';

import getColorFromInitials from '../../utils/getColorFromInitials';
import styles from './Avatar.module.scss';

const Avatar = ({url, firstName, lastName}) => {
  if (url) {
    return (
      <Image src={url} alt={`${firstName} ${lastName}`} width="32" height="32" circular centered />
    );
  }

  const initials = [firstName, lastName]
    .map(upperCase)
    .map(text => get(text, '0', ''))
    .join('');

  const color = getColorFromInitials(initials);

  return !!(firstName || lastName) && (
    <div className={styles.placeholder} style={{backgroundColor: color}}>
      {initials}
    </div>
  );
};

Avatar.defaultProps = {
  firstName: '',
  lastName: '',
  url: null,
};

Avatar.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  url: PropTypes.string,
};

export default Avatar;
