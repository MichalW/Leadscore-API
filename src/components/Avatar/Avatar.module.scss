@import '../../styles/colors';

.placeholder {
  border-radius: 16px;
  color: $color-white;
  font-size: 12px;
  height: 32px;
  line-height: 32px;
  margin: 0 auto;
  text-align: center;
  width: 32px;
}
