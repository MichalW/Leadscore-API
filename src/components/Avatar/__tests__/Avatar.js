import React from 'react';
import toJson from 'enzyme-to-json';
import {Image} from 'semantic-ui-react';
import {shallow} from 'enzyme';

import Avatar from '../Avatar';

it('should render Avatar with url', () => {
  const props = {
    url: 'https://example.com/avatar.png',
  };

  const avatar = shallow(<Avatar {...props} />);

  expect(avatar.find(Image).exists()).toBe(true);
  expect(toJson(avatar)).toMatchSnapshot();
});


it('should render Avatar without url', () => {
  const props = {
    firstName: 'Example',
    lastName: 'User',
  };

  const avatar = shallow(<Avatar {...props} />);

  expect(avatar.find(Image).exists()).toBe(false);
  expect(toJson(avatar)).toMatchSnapshot();
});
