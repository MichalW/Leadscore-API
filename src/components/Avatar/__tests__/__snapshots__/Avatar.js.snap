// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Avatar with url 1`] = `
<Image
  alt=" "
  as="img"
  centered={true}
  circular={true}
  height="32"
  src="https://example.com/avatar.png"
  ui={true}
  width="32"
/>
`;

exports[`should render Avatar without url 1`] = `
<div
  className="placeholder"
  style={
    Object {
      "backgroundColor": "rgb(183,55,71)",
    }
  }
>
  EU
</div>
`;
